zerver/lib/actions.py

```diff
@@ -526,23 +572,28 @@ def create_mirror_user_if_needed(realm, email, email_to_fullname):
             return get_user_profile_by_email(email)
 
 def log_message(message):
+    # type: (Message) -> None
     if not message.sending_client.name.startswith("test:"):
         log_event(message.to_log_dict())
 
 def always_push_notify(user):
+    # type: (UserProfile) -> bool
     # robinhood.io asked to get push notifications for **all** notifyable
     # messages, regardless of idle status
     return user.realm.domain in ['robinhood.io']
 
 # Helper function. Defaults here are overriden by those set in do_send_messages
 def do_send_message(message, rendered_content = None, no_log = False, stream = None, local_id = None):
+    # type: (Union[int, Message], text_type, bool, Optional[Stream], int) -> int
```

Comment:
Yes `Optional[text_type]` is better and the last argument `local_id`'s type is better as `Optional[int]`. I'll change it.